### PR TITLE
HPO in Hugging Face Trainer benchmark

### PR DESCRIPTION
**Pull request by mseeger**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: benchmarking/nursery/transformer_glue/hpo_main.py [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+    optimizer = args.optimizer
+
+    # Path to training script. We also need to specify the names of metrics reported
+    # back from this script
+    entry_point = "./run_glue_modified.py"
+    metric = "eval_" + TASK2METRICSMODE[dataset]["metric"]
+    mode = TASK2METRICSMODE[dataset]["mode"]
+    resource_attribute = "epoch"
+
+    # [1]
```
Comment (aaronkl) [inline benchmarking/nursery/transformer_glue/hpo_main.py:112]: dead comment

File: benchmarking/nursery/transformer_glue/hpo_main.py [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+        "max_seq_length": 128,
+        "output_dir": "tmp/" + dataset,
+        "evaluation_strategy": "epoch",
+        "save_strategy": "no",  # change to "epoch" if checkpoints are needed!
+        "save_total_limit": 1,
+    }
+
+    config_space = {**hyperparameter_space, **fixed_parameters}
+
+    # [2]
```
Comment (aaronkl) [inline benchmarking/nursery/transformer_glue/hpo_main.py:140]: dead comment

File: benchmarking/nursery/transformer_glue/hpo_main.py [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+    # This is the default configuration provided by Hugging Face. It will always
+    # be evaluated first
+    default_configuration = {
+        "learning_rate": 2e-5,
+        "per_device_train_batch_size": 32,
+        "warmup_ratio": 0.0,
+    }
+
+    # [3]
```
Comment (aaronkl) [inline benchmarking/nursery/transformer_glue/hpo_main.py:149]: same

File: benchmarking/nursery/transformer_glue/launch_remote.py [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+            "choose_model": int(choose_model),
+            "seed": seed,
+        }
+
+        # Pass Syne Tune sources as dependencies
+        source_dir = str(Path(__file__).parent)
+        entry_point = "hpo_main.py"
+        dependencies = syne_tune.__path__ + [source_dir]
+        # Latest PyTorch version (1.10):
+        est = PyTorch(
```
Comment (aaronkl) [inline benchmarking/nursery/transformer_glue/launch_remote.py:94]: Why not using the HuggingFace container here?

**Review by aaronkl — APPROVED** [review]
LGTM. Nitpick: could we change the directory name to, for example, '/benchmarking/nursery/fine_tuning_glue/'

File: benchmarking/nursery/transformer_glue/launch_remote.py (hunk 4 above)
Comment (mseeger) [reply benchmarking/nursery/transformer_glue/launch_remote.py:94]: Good point. Should we? I tested it all with PyTorch framework.

**Review by aaronkl — APPROVED** [review]

**Pull request merged**